### PR TITLE
Added `config init` command

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR

- Update export message for kubeconfig
- Init config sandbox for sandbox

```bash
$ flytectl config  --help
Usage:
  flytectl config [command]

Available Commands:
  discover    Searches for a config in one of the default search paths.
  init        Generates flytectl config file in the user's home directory.
  validate    Validates the loaded config.

Flags:
      --file stringArray   Passes the config file to load.
                           If empty, it'll first search for the config file path then, if found, will load config from there.
  -h, --help               help for config

# if the doesn't config exist
$ flytectl config init  
#  if the config exist
$ flytectl config init  
This action will overwrite an existing config file at [/Users/yuvraj/.flyte/config.yaml]. Do you want to continue? [y/n]: Y
$ cat ~/.flyte/config.yaml
admin:
  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
  endpoint: dns:///localhost:30081
  authType: Pkce
  insecure: true
logger:
  show-source: true
  level: 1
storage:
  connection:
    access-key: minio
    auth-type: accesskey
    disable-ssl: true
    endpoint: http://localhost:30084
    region: us-east-1
    secret-key: miniostorage
  type: minio
  container: "my-s3-bucket"
  enable-multicontainer: true
```

- Init config for the remote cluster

```bash
$ flytectl config init   --host=flyte.myexample.com
Use the arrow keys to navigate: ↓ ↑ → ← 
? Select Storage Provider: 
  ▸ S3
    GCS
This action will overwrite an existing config file at [/Users/yuvraj/.flyte/config.yaml]. Do you want to continue? [y/n]: Y
Init flytectl config for the remote cluster, Please update your storage config in ~/.flyte/config.yaml. Learn more about the config here https://docs.flyte.org/projects/flytectl/en/latest/index.html#configure
$ cat ~/.flyte/config.yaml
admin:
  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
  endpoint: dns:///flyte.myexample.com
  authType: Pkce
  insecure: true
logger:
  show-source: true
  level: 0
storage:
  type: stow	
  stow:
    kind: s3
    config:
      auth_type: iam
      region: <replace> # Example: us-east-2
  container: <replace> # Example my-bucket. Flyte k8s cluster / service account for execution should have read access to this bucket


$ flytectl config init  --host=flyte.myexample.com
Use the arrow keys to navigate: ↓ ↑ → ← 
? Select Storage Provider: 
    S3
 ▸ GCS
This action will overwrite an existing config file at [/Users/yuvraj/.flyte/config.yaml]. Do you want to continue? [y/n]: Y
Init flytectl config for the remote cluster, Please update your storage config in ~/.flyte/config.yaml. Learn more about the config here https://docs.flyte.org/projects/flytectl/en/latest/index.html#configure
$ cat ~/.flyte/config.yaml
admin:
  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
  endpoint: dns:///flyte.myexample.com
  authType: Pkce
  insecure: true
logger:
  show-source: true
  level: 0
storage:
  type: stow	
  stow:
    kind: s3
    config:
      json: ""
      project_id: <replace-me> # TODO: replace <project-id> with the GCP project ID
      scopes: https://www.googleapis.com/auth/devstorage.read_write
  container: <replace> # Example my-bucket. Flyte k8s cluster / service account for execution should have read access to this bucket
```

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: pkg/util/util.go [hunk 1]
```diff
@@ -11,6 +11,27 @@ type githubversion struct {
...
+    access-key: minio
+    auth-type: accesskey
+    disable-ssl: true
+    endpoint: http://localhost:30084
+    region: us-east-1
+    secret-key: miniostorage
+  type: minio
+  container: "my-s3-bucket"
+  enable-multicontainer: true
+`
```
Comment (pmahindrakar-oss) [inline pkg/util/util.go:33]: can we not use the config.yaml as the template instead of keeping both these.
I find people have to make changes in both the places which might not be likeable

File: cmd/config.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	"io"
+	"os"
+
+	cmdUtil "github.com/flyteorg/flytectl/pkg/commandutils"
+	"github.com/flyteorg/flytectl/pkg/docker"
+	f "github.com/flyteorg/flytectl/pkg/filesystemutils"
+	"github.com/flyteorg/flytectl/pkg/util"
+	"github.com/spf13/cobra"
+)
+
```
Comment (pmahindrakar-oss) [inline cmd/config.go:13]: Missing command docs.

Comment (evalsocket) [reply cmd/config.go:13]: done

File: pkg/util/util.go (hunk 1 above)
Comment (evalsocket) [reply pkg/util/util.go:33]: I think we can remove config.yaml from the repo and use init command

File: cmd/config.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+// configCmd represents the config init command
+var configCmd = &cobra.Command{
+	Use:   "init-config",
```
[the remark this answers is not in the document]
Comment (EngHabu) [reply cmd/config.go:16]: We might later add `flytectl init python-module` or others ;)

[the remark this answers is not in the document]
Comment (EngHabu) [reply cmd/config.go:16]: CC @kumare3

File: cmd/config.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+ bin/flytectl init-config
+Usage`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return initFlytectlConfig(os.Stdin)
+	},
+}
+
+func initFlytectlConfig(reader io.Reader) error {
+	if err := docker.SetupFlyteDir(); err != nil {
```
Comment (EngHabu) [inline cmd/config.go:30]: Why is `SetupFlyteDir` is on `docker` ?

File: cmd/config.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+		return err
+	}
+	if cmdUtil.AskForConfirmation("Are you sure ? It will overwrite the default config from ~/.flyte/config.yaml", reader) {
```
Comment (EngHabu) [inline cmd/config.go:33]: Let's not pop up the question unless the file exist... this will be weird for new installations

File: cmd/config.go [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		return util.WriteIntoFile([]byte(util.ConfigTemplate), f.FilePathJoin(f.UserHomeDir(), ".flyte", "config.yaml"))
```
Comment (EngHabu) [inline cmd/config.go:34]: The goal of this command is to allow users to setup config given an endpoint...
examples:

```
# Setup default config to sandbox
flytectl init config

# Setup config that talks to flyte.mycompany.com
flytectl init config --host flyte.mycompany.com

# Setup config with custom storage config
flytectl init config --host flyte.mycompany.com --storage.type=s3 --storage.container=my-bucket 
```

File: cmd/init/config.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+	cmdCore "github.com/flyteorg/flytectl/cmd/core"
+	cmdUtil "github.com/flyteorg/flytectl/pkg/commandutils"
+	"github.com/flyteorg/flytectl/pkg/util"
+)
+
+const (
+	initConfigCmdShort = "Used for generating config template."
+	initConfigCmdLong  = `init config will create a config in flyte directory i.e ~/.flyte
+Generate sandbox config.
```
Comment (pmahindrakar-oss) [inline cmd/init/config.go:19]: Can we link to what does it mean to have a sandbox https://docs.flyte.org/en/latest/deployment/sandbox.html
and also configuration related to it.

File: cmd/init/config.go [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+	
+::
+
+ bin/flytectl init config 
+
+Generate remote cluster config.
```
Comment (pmahindrakar-oss) [inline cmd/init/config.go:25]: Add a Link to remote deployment https://docs.flyte.org/en/latest/deployment/index.html

File: pkg/util/util.go [hunk 9]
```diff
@@ -5,12 +5,50 @@ import (
...
 
+const (
+	ConfigTemplate = `admin:
+  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
+  endpoint: {{.Host}}
+  authType: Pkce
+  insecure: {{.Insecure}}
+logger:
+  show-source: true
+  level: 1`
```
Comment (pmahindrakar-oss) [inline pkg/util/util.go:26]: keep the level : 0

File: cmd/init/config.go [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+	}
+	spec := util.ConfigTemplateSpec{
+		Host:     "dns:///localhost:30081",
+		Insecure: true,
+	}
+	configTemplate := util.ConfigTemplate + util.StorageTemplate
+
+	if len(initConfig.DefaultConfig.Host) > 0 {
+		spec.Host = fmt.Sprintf("dns:///%v", initConfig.DefaultConfig.Host)
+		spec.Insecure = false
```
Comment (pmahindrakar-oss) [inline cmd/init/config.go:49]: lets keep insecure to be true

File: cmd/init/config.go [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+		configTemplate = util.ConfigTemplate
+	}
+	var _err error
+	if _, err := os.Stat(util.ConfigFile); os.IsNotExist(err) {
+		_err = util.SetupConfig(configTemplate, util.ConfigFile, spec)
+	} else {
+		if cmdUtil.AskForConfirmation("Are you sure ? It will overwrite the default config ~/.flyte/config.yaml", reader) {
```
Comment (pmahindrakar-oss) [inline cmd/init/config.go:56]: ~/.flyte/config.yaml to util.ConfigFile

File: cmd/config/subcommand/init/config_flags.go [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+package init
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+)
+
+// GetPFlagSet will return strongly types pflags for all fields in Config and its nested types. The format of the
+// flags is json-name.json-sub-name... etc.
```
Comment (pmahindrakar-oss) [inline cmd/config/subcommand/init/config_flags.go:13]: Did you manually remove the generated functions ?

File: pkg/util/util.go [hunk 13]
```diff
@@ -5,12 +5,50 @@ import (
...
+
+	f "github.com/flyteorg/flytectl/pkg/filesystemutils"
 )
 
 type githubversion struct {
 	TagName string `json:"tag_name"`
 }
 
+const (
+	ConfigTemplate = `admin:
```
Comment (pmahindrakar-oss) [inline pkg/util/util.go:19]: nit : AdminConfigTemplate

File: pkg/util/util.go [hunk 14]
```diff
@@ -5,12 +5,50 @@ import (
...
+  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
+  endpoint: {{.Host}}
+  authType: Pkce
+  insecure: {{.Insecure}}
+logger:
+  show-source: true
+  level: 1`
+	StorageTemplate = `
```
Comment (pmahindrakar-oss) [inline pkg/util/util.go:27]: nit : StorageConfigTemplate

File: pkg/util/util.go [hunk 15]
```diff
@@ -5,12 +5,50 @@ import (
...
+    secret-key: miniostorage
+  type: minio
+  container: "my-s3-bucket"
+  enable-multicontainer: true`
+)
+
+type ConfigTemplateSpec struct {
+	Host     string
+	Insecure bool
+}
```
Comment (pmahindrakar-oss) [inline pkg/util/util.go:44]: ConfigTemplateValuesSpec

File: cmd/sandbox/start.go [hunk 16]
```diff
@@ -57,11 +59,16 @@ func startSandboxCluster(ctx context.Context, args []string, cmdCtx cmdCore.Comm
...
+	if err := util.SetupFlyteDir(); err != nil {
 		return nil, err
 	}
 
-	if err := docker.GetFlyteSandboxConfig(); err != nil {
+	spec := util.ConfigTemplateSpec{
+		Host:     "localhost:30081",
+		Insecure: true,
+	}
+	configTemplate := util.ConfigTemplate + util.StorageTemplate
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/start.go:70]: move this to a function called createCombinedTemplate and return the concatenated templates.

File: cmd/sandbox/start.go [hunk 17]
```diff
@@ -57,11 +59,16 @@ func startSandboxCluster(ctx context.Context, args []string, cmdCtx cmdCore.Comm
 
 func startSandbox(ctx context.Context, cli docker.Docker, reader io.Reader) (*bufio.Scanner, error) {
 	fmt.Printf("%v Bootstrapping a brand new flyte cluster... %v %v\n", emoji.FactoryWorker, emoji.Hammer, emoji.Wrench)
-	if err := docker.SetupFlyteDir(); err != nil {
+	if err := util.SetupFlyteDir(); err != nil {
 		return nil, err
 	}
 
-	if err := docker.GetFlyteSandboxConfig(); err != nil {
+	spec := util.ConfigTemplateSpec{
```
Comment (pmahindrakar-oss) [inline cmd/sandbox/start.go:66]: nit : spec => templateValues

File: pkg/util/util.go [hunk 18]
```diff
@@ -5,12 +5,50 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"os"
+	"text/template"
+
+	f "github.com/flyteorg/flytectl/pkg/filesystemutils"
```
Comment (pmahindrakar-oss) [inline pkg/util/util.go:11]: Instead of package name util, can we create config_util instead with function related to config creation there

File: cmd/init/init.go [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+	
+::
+
+ bin/flytectl init config 
+
+Generate remote cluster config. 
+	
+::
+
+ bin/flytectl init config --host="flyte.myexample.com"
```
Comment (pmahindrakar-oss) [inline cmd/init/init.go:24]: These docs seems to be duplicated.

File: cmd/init/config.go (hunk 11 above)
Comment (evalsocket) [reply cmd/init/config.go:56]: meaning ?

File: cmd/config/subcommand/init/config_flags.go (hunk 12 above)
Comment (evalsocket) [reply cmd/config/subcommand/init/config_flags.go:13]: yes

File: cmd/init/init.go (hunk 19 above)
Comment (evalsocket) [reply cmd/init/init.go:24]: It's only one command

File: pkg/util/util.go (hunk 18 above)
Comment (evalsocket) [reply pkg/util/util.go:11]: no-sandbox also use these function

File: cmd/config/subcommand/init/init_flags.go [hunk 20]
```diff
@@ -0,0 +1,11 @@
+package init
+
+//go:generate pflags Config --default-var DefaultConfig
+var (
+	DefaultConfig = &Config{}
+)
+
+//Configs
+type Config struct {
+	Host string `json:"host" pflag:", Endpoint of flyte admin"`
```
Comment (EngHabu) [inline cmd/config/subcommand/init/init_flags.go:10]: nit:
```suggestion
	Host string `json:"host" pflag:",Endpoint of flyte admin"`
```

File: cmd/init/init.go [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	initConfig "github.com/flyteorg/flytectl/cmd/config/subcommand/init"
+	cmdcore "github.com/flyteorg/flytectl/cmd/core"
+
+	"github.com/spf13/cobra"
+)
+
+// Long descriptions are whitespace sensitive when generating docs using sphinx.
+const (
+	initCmdShort = `Used for generating config template.`
```
Comment (EngHabu) [inline cmd/init/init.go:12]:
```suggestion
	initCmdShort = `Generates flytectl config file in the user's home directory.`
```

File: cmd/init/init.go [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+	initCmdLong  = `init config will create a config in flyte directory i.e ~/.flyte
```
Comment (EngHabu) [inline cmd/init/init.go:13]:
```suggestion
	initCmdLong  = `Creates a flytectl config file in flyte directory i.e ~/.flyte
```

File: cmd/init/init.go [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+
+::
+
+ bin/flytectl init config 
+
+Generate remote cluster config. Read more about the remote deployment https://docs.flyte.org/en/latest/deployment/index.html
+	
+::
+
+ bin/flytectl init config --host="flyte.myexample.com"
```
Comment (EngHabu) [inline cmd/init/init.go:25]:
```suggestion
 bin/flytectl init config --host=flyte.myexample.com
```

File: pkg/util/util.go [hunk 24]
```diff
@@ -5,12 +5,81 @@ import (
...
+    auth-type: accesskey
+    disable-ssl: true
+    endpoint: http://localhost:30084
+    region: us-east-1
+    secret-key: miniostorage
+  type: minio
+  container: "my-s3-bucket"
+  enable-multicontainer: true`
+	StorageS3ConfigTemplate = `
+#storage:
```
Comment (EngHabu) [inline pkg/util/util.go:40]: Why is this commented? `#`

File: cmd/config/subcommand/init/init_flags.go [hunk 25]
```diff
@@ -0,0 +1,11 @@
+package init
+
+//go:generate pflags Config --default-var DefaultConfig
```
Comment (EngHabu) [inline cmd/config/subcommand/init/init_flags.go:3]: You need the --bind-to-default (can't remember the exact name) if you want to bind .DefaultConfig.Host to the value the user provided

File: cmd/init/config.go [hunk 26]
```diff
@@ -0,0 +1,63 @@
...
+
+	templateValues := util.ConfigTemplateValuesSpec{
+		Host:     "dns:///localhost:30081",
+		Insecure: true,
+		Template: util.GetSandboxTemplate(),
+	}
+
+	if len(initConfig.DefaultConfig.Host) > 0 {
+		templateValues.Host = fmt.Sprintf("dns:///%v", initConfig.DefaultConfig.Host)
+		templateValues.Insecure = true
```
Comment (EngHabu) [inline cmd/init/config.go:37]: We shouldn't assume Insecure all the time... have you tested this with `demo.nuclyde.io` ? you will probably need to add `--insecure` support as well...

File: cmd/init/config.go [hunk 27]
```diff
@@ -0,0 +1,63 @@
...
+		}
+		if result == "GCS" {
+			templateValues.Template = util.GetGoogleCloudTemplate()
+		}
+	}
+	var _err error
+	if _, err := os.Stat(util.ConfigFile); os.IsNotExist(err) {
+		_err = util.SetupConfig(util.ConfigFile, templateValues)
+	} else {
+		if cmdUtil.AskForConfirmation(fmt.Sprintf("Are you sure ? It will overwrite the default config %v", util.ConfigFile), reader) {
```
Comment (EngHabu) [inline cmd/init/config.go:51]:
```suggestion
		if cmdUtil.AskForConfirmation(fmt.Sprintf("This action will overwrite an existing config file at [%s]. Do you want to continue?", util.ConfigFile), reader) {
```

File: pkg/util/util.go [hunk 28]
```diff
@@ -5,12 +5,85 @@ import (
...
+      json: ""
+      project_id: <replace-me> # TODO: replace <project-id> with the GCP project ID
+      scopes: https://www.googleapis.com/auth/devstorage.read_write
+  container: <replace> # Example my-bucket. Flyte k8s cluster / service account for execution should have read access to this bucket`
+)
+
+type ConfigTemplateValuesSpec struct {
+	Host     string
+	Insecure bool
+	Template string
```
Comment (pmahindrakar-oss) [inline pkg/util/util.go:63]: Template itself being part of the value spec seems odds,

File: pkg/util/util.go [hunk 29]
```diff
@@ -41,3 +114,39 @@ func WriteIntoFile(data []byte, file string) error {
...
+// SetupFlyteDir will create .flyte dir if not exist
+func SetupFlyteDir() error {
+	if err := os.MkdirAll(f.FilePathJoin(f.UserHomeDir(), ".flyte"), 0755); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetupConfig download the flyte sandbox config
+func SetupConfig(filename string, templateValue ConfigTemplateValuesSpec) error {
```
Comment (pmahindrakar-oss) [inline pkg/util/util.go:127]: pass templateValue and template separately and not part of the same structure

File: pkg/docker/docker_util.go [hunk 30]
```diff
@@ -176,12 +142,21 @@ func ReadLogs(ctx context.Context, cli Docker, id string) (*bufio.Scanner, error
...
 	for reader.Scan() {
 		if strings.Contains(reader.Text(), message) {
+			kubeconfig := strings.Join([]string{
+				"$KUBECONFIG",
+				f.FilePathJoin(f.UserHomeDir(), ".kube", "config"),
+				Kubeconfig,
+			}, ":")
+			os.Setenv("KUBECONFIG", kubeconfig)
+			os.Setenv("FLYTECTL_CONFIG", util.FlytectlConfig)
+
```
Comment (pmahindrakar-oss) [inline pkg/docker/docker_util.go:152]: Whats the purpose of doing this when we asking the user to do it by echoing the export command

File: pkg/util/util.go (hunk 28 above)
Comment (evalsocket) [reply pkg/util/util.go:63]: no I can change  the variable name and that would be ok

**Review by pmahindrakar-oss — DISMISSED** [review]

File: pkg/configutil/configutil.go [hunk 31]
```diff
@@ -0,0 +1,104 @@
...
+storage:
+  type: stow	
+  stow:
+    kind: s3
+    config:
+      json: ""
+      project_id: <replace-me> # TODO: replace <project-id> with the GCP project ID
+      scopes: https://www.googleapis.com/auth/devstorage.read_write
+  container: <replace> # Example my-bucket. Flyte k8s cluster / service account for execution should have read access to this bucket`
+)
```
Comment (pmahindrakar-oss) [inline pkg/configutil/configutil.go:50]: Can you verify if these actually work.
I have heard from two users who have complained about the aws one which they tried.
```
Error:
1 error(s) decoding:
* '' has invalid keys: config, kind
ERRO[0000]
1 error(s) decoding:
* '' has invalid keys: config, kind  src="main.go:13"
this I got after adding storage in config.yaml
```

Comment (pmahindrakar-oss) [reply pkg/configutil/configutil.go:50]: Lets also update the docs https://docs.flyte.org/projects/flytectl/en/stable/ with right config

Comment (evalsocket) [reply pkg/configutil/configutil.go:50]: yes there is no issue with parsing

File: pkg/util/util.go [hunk 32]
```diff
@@ -41,3 +44,11 @@ func WriteIntoFile(data []byte, file string) error {
 	}
 	return nil
 }
+
+// SetupFlyteDir will create .flyte dir if not exist
+func SetupFlyteDir() error {
+	if err := os.MkdirAll(f.FilePathJoin(f.UserHomeDir(), ".flyte"), 0755); err != nil {
```
Comment (EngHabu) [inline pkg/util/util.go:50]: Is that equivalent to `os.ModePerm` ?

File: cmd/configuration/configuration.go [hunk 33]
```diff
@@ -0,0 +1,97 @@
...
+	templateValues := configutil.ConfigTemplateSpec{
+		Host:     "dns:///localhost:30081",
+		Insecure: initConfig.DefaultConfig.Insecure,
+	}
+	templateStr := configutil.GetSandboxTemplate()
+
+	if len(initConfig.DefaultConfig.Host) > 0 {
+		templateValues.Host = fmt.Sprintf("dns:///%v", initConfig.DefaultConfig.Host)
+		templateStr = configutil.GetAWSCloudTemplate()
+		_, result, err := prompt.Run()
```
[the remark this answers is not in the document]
Comment (evalsocket) [reply cmd/configuration/configuration.go:73]: It run as interactive mode

[the remark this answers is not in the document]
Comment (EngHabu) [reply cmd/configuration/configuration.go:73]: Should this be another flag and if not set, we use the prompt then?
```suggestion
                var result string
                var err error
                if len(initConfig.DefaultConfig.StorageType) == 0 {
  		  _, result, err = prompt.Run()
                }
```

File: cmd/configuration/configuration.go [hunk 34]
```diff
@@ -0,0 +1,108 @@
...
+Generate remote cluster config. Read more about the remote deployment https://docs.flyte.org/en/latest/deployment/index.html
+	
+::
+
+ bin/flytectl configuration config --host=flyte.myexample.com
+	
+Generate flytectl config with a specific storage type
+::
+
+ bin/flytectl configuration config --host=flyte.myexample.com --storage=gcs
```
Comment (pmahindrakar-oss) [inline cmd/configuration/configuration.go:43]: instead of having him select the storage here and also in prompt , i would prefer to do it only once to avoid user providing gcs in command line args and then in prompt selecting something else
We can keep this flag as boolean and have all type of storage being available in the prompt. 

Over here we can add an example with how the prompt would show up in aws or gcs case.

Comment (evalsocket) [reply cmd/configuration/configuration.go:43]: it will only prompt if user didn't pass flag for storage

**Review by pmahindrakar-oss — APPROVED** [review]
